projects/igniteui-angular/src/lib/grid/cell.component.ts

```diff
@@ -283,6 +283,9 @@ export class IgxGridCellComponent implements OnInit, AfterViewInit {
      * @memberof IgxGridCellComponent
      */
     set inEditMode(value: boolean) {
+        if (this.grid.rowEditable && this.row.deleted) {
```

Comment:
Can't the row get deleted via API even if there's no row editing enabled?

Reply:
No, there isn't a deleted row, when row editing is disabled and yes the check for rowEditable is not needed. Thanks!

Reply:
I'm pretty sure that there is:

https://github.com/IgniteUI/igniteui-angular/blob/master/projects/igniteui-angular/src/lib/grid/grid.component.ts#L3078

Reply:
Right! Please ignore the first part of my previous answer. The correct thing is that only the `this.row.deleted` check is enough.